application/libraries/Librivox_search.php

```diff
@@ -77,16 +77,31 @@ function advanced_title_search($params)
 		$recorded_language = (int)$params['recorded_language'];
 		if ($recorded_language !== 0)
 		{
-			//$language = ' AND p.language_id = ' . $params['recorded_language'] ;
+			if (empty($params['title']) and empty($params['author']) and empty($params['reader']))
+			{
+				// If we're just browsing, we can remove some redundant entries to make the results more compact.
 
-			// Projects having at least one section in that language.
-			$section_project_ids = $this->_get_projects_by_section_language($recorded_language);
-			//$language_clause = ' AND p.id IN (' . implode(',', $section_project_ids) . ') ';
+				// Projects: only listed if they are primarily in this language.
+				$language_clause = ' AND p.language_id = ' . $recorded_language;
 
-			$language_clause = ' AND ( p.language_id = ' . $recorded_language . ' OR p.id IN (' . implode(',', $section_project_ids) . ') ) ';
-			$section_language_clause = '
-				AND ( p.language_id = ' . $recorded_language . '
-					OR st.section_language_id = ' . $recorded_language . ') ';
+				// Sections: only (individually) listed if they are in this language, but from a project we wouldn't list above.
+				$section_language_clause = '
+					AND p.language_id <> ' . $recorded_language . '
+					AND st.section_language_id = ' . $recorded_language;
+			}
+			else
+			{
+				// If we're searching by specifics, we should cast a wide net for potential matches.
+
+				// Projects: search among all that have any sections in this language.
+				$section_project_ids = $this->_get_projects_by_section_language($recorded_language);
+				$language_clause = ' AND ( p.language_id = ' . $recorded_language . ' OR p.id IN (' . implode(',', $section_project_ids) . ') ) ';
```

Comment:
This is probably fine, but I want to be mindful of performance traps, what with neither of us being DBAs.

Reading [1], "This means IN is very quick if the IN value list consists entirely of constants." Obviously `$section_project_ids` can't be constants, but I wonder if we'd gain much by `CAST`ing them to the same type as `language_id`...

[1] https://mariadb.com/kb/en/in/